app/services/data_processor.py

```
@@ -0,0 +1,10 @@
+def process_data(data):
+    processed = ""
+    for char in data:
+        if char.isalpha():
+            processed += char.lower()
+        else:
+            processed += char
+    for i in range(100):
```

Ruff (0.8.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)